src/toast/tests/noise.py

```diff
@@ -39,15 +43,27 @@ def test_noise_hdf5(self):
         nse = Noise(
             detnames,
             {x: u.Quantity(freqs, u.Hz) for x in streams},
-            {x: u.Quantity(np.ones(len(freqs)), u.K ** 2 * u.second) for x in streams},
+            {x: u.Quantity(np.ones(len(freqs)), u.K**2 * u.second) for x in streams},
             mixmatrix=mix,
         )
 
         nse_file = os.path.join(self.outdir, "noise.h5")
 
-        nse.save_hdf5(nse_file, comm=self.comm)
+        hf = hdf5_open(nse_file, "w", comm=self.comm)
```

Comment:
This is another place where we could move boiler plate into the `Noise` class.

Reply:
As discussed offline, having all of the lower level classes only work with `h5py.Group` objects relieves those low-level functions from having to handle all the different types of possible cases.  The downside is that it adds more lines to the higher level code to open the HDF5 file and close it if the local process has a valid handle.  I'll add a small context manager around hdf5_open() to clean up some of this.